### PR TITLE
2. Get the component to show up in the user activity tab. Eg /users/3/actions

### PR DESCRIPTION
**Pull request by FaithMo**
Related issues: [2-Get-the-component-to-show-up-in-the-user-activity-tab-Eg-users-3-actions](https://www.notion.so/2-Get-the-component-to-show-up-in-the-user-activity-tab-Eg-users-3-actions-2967c2e81b51414386eab1a29cd21da0)

## Description:

What are you up to? Fill us in :)

## Screenshots/videos
image

## I solemnly swear that:

- [ ] My code follows the style guidelines of this project
- [ ] I have merged the develop branch into my branch and fixed any merge conflicts
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: frontend/src/components/regions/UserActions/index.jsx [hunk 1]
```diff
@@ -196,44 +187,52 @@ function UserActionsUnconnected({
...
+  console.log(orderedDates2);
 
-  let actionLogByDate = {};
-  actionLog.forEach((o) => {
-    const date = o.dateStr;
-    if (orderedDates.indexOf(date) === -1) orderedDates.push(date);
-    actionLogByDate[date] = actionLogByDate[date] || [];
-    actionLogByDate[date].push(o);
-  });
+  console.log("bfore", activityLogEntries);
```
Comment (Ngoakor12) [inline frontend/src/components/regions/UserActions/index.jsx]: don't forget to remove console.logs

File: frontend/src/components/regions/UserActions/index.jsx [hunk 2]
```diff
@@ -196,44 +187,52 @@ function UserActionsUnconnected({
...
   const eventTypesWithColors = matchEventTypesWithColors({
     eventTypes,
     eventTypeColors,
   });
 
   activityLogEntries = mapData({ eventTypesWithColors, activityLogEntries });
+  console.log("after", activityLogEntries);
```
Comment (Ngoakor12) [inline frontend/src/components/regions/UserActions/index.jsx]: remove console.log

File: frontend/src/components/regions/UserActions/index.jsx [hunk 3]
```diff
@@ -54,6 +91,19 @@ function UserActionsUnconnected({
     fetchUserBurndownStats({ userId });
   }, [fetchCardCompletions, fetchUserBurndownStats, userId]);
 
+  useEffect(() => {
+    fetchActivityLogEntries({
+      actorUser: userId,
+      page: 3,
```
Comment (Ngoakor12) [inline frontend/src/components/regions/UserActions/index.jsx]: should we not start fetching from the first page?

**Review by Ngoakor12 — CHANGES_REQUESTED** [review]
Well done on updating the redux api app for activity log entries, it looks good to me.

- **The storybook for the activityLog is broken**, most likey due to the change in the API I think? Maybe the data in the `activityLogListing.json` file needs to be updated?
image

- The activityLogEntries API only fetches one page at a time unlike the current activity log listing api in production, so **some pagination is likely needed**(even though I'm not sure how at this moment), maybe copy how the current activity log listing fetches multiple pages via scrolling to the end?

- Try and address the warnings generated if you're able to, the one related to the media query styling can be solved by making sure that the comment is on the same line as as the one  causing the warning. Turning off "format on save" should help the stay on the same line
image

File: frontend/src/components/regions/UserActions/index.jsx (hunk 3 above)
Comment (FaithMo) [reply frontend/src/components/regions/UserActions/index.jsx]: i forgot to change it back t 1, im on it.

File: frontend/src/components/regions/UserActions/ActivityLog.jsx [hunk 4]
```diff
@@ -3,10 +3,15 @@ import Typography from "@material-ui/core/Typography";
...
 import Chip from "@material-ui/core/Chip";
-import Grid from "@material-ui/core/Grid";
 import { makeStyles } from "@material-ui/core/styles";
 
 const useStyles = makeStyles((theme) => ({
+  paper: {
+    width: "50rem",
+    ["@media (max-width:620px)"]: {
+      fontSize: "1rem",
+    },
```
Comment (Ngoakor12) [inline frontend/src/components/regions/UserActions/ActivityLog.jsx]: One tiny thing, this line generates a warning. Here's how I fixed it:
image

**Review by Ngoakor12 — CHANGES_REQUESTED** [review]

File: frontend/src/components/regions/UserActions/index.jsx [hunk 5]
```diff
@@ -87,6 +145,11 @@ function UserActionsUnconnected({
       const nextCardPage = lastCompletedCardsPage.requestData.page + 1;
       fetchCardCompletions({ page: nextCardPage, assigneeUserId: userId });
     }
+    if (latestActivityLogPage.responseData.results.length > 0) {
+      const nextCardPage = latestActivityLogPage.requestData.page + 1;
+      console.log({ page: nextCardPage, actorUser: userId });
```
Comment (Ngoakor12) [inline frontend/src/components/regions/UserActions/index.jsx]: you also forgot to remove this console.log

**Review by Ngoakor12 — CHANGES_REQUESTED** [review]

**Review by Ngoakor12 — APPROVED** [review]

**Comment by KhalidUmuzi** [comment]
I think as a part of this, we should remove the old DayLog.jsx. It is going to be replaced with this new ActivityLog.jsx so we might as well get rid of it now

File: frontend/src/components/regions/UserActions/ActivityLog.jsx [hunk 6]
```diff
@@ -41,26 +39,27 @@ export default function ActivityLog({ eventList, sortedTimestampArray }) {
   return (
     <React.Fragment>
       {sortedTimestampArray.map((timestamp) => (
```
Comment (KhalidUmuzi) [inline frontend/src/components/regions/UserActions/ActivityLog.jsx]: The name of this array could be better. We did this as an example but if i were to look at it now, I'm thinking this array is an array of timestamps that have been sorted and nothing else.

File: frontend/src/components/regions/UserActions/index.jsx [hunk 7]
```diff
@@ -22,17 +23,54 @@ const days = [
...
   authedUserId,
   projectReviews,
   cardSummaries,
+  activityLogEntries,
   fetchProjectReviewsPages,
   fetchCardCompletions,
   userBurndownStats,
   fetchUserBurndownStats,
+  fetchActivityLogEntries,
+  eventTypes,
```
Comment (KhalidUmuzi) [inline frontend/src/components/regions/UserActions/index.jsx]: small placement thing here. place the mapStateToProps variables at the top to keep consistency

File: frontend/src/components/regions/UserActions/index.jsx [hunk 8]
```diff
@@ -22,17 +23,54 @@ const days = [
...
+    arr.push({
+      id: eventType.id,
+      eventName: eventType.name,
+      eventColor: eventTypeColors[eventType.name],
+    });
+  });
+  return arr;
+};
+
+function mapData({ eventTypesWithColors, activityLogEntries }) {
```
Comment (KhalidUmuzi) [inline frontend/src/components/regions/UserActions/index.jsx]: This is actually a VERY useful function that I can see being used in other places and not just here. In an effort to    avoid non-dry code, could we perhaps take this function, make it universal and place it in a utility file of some sort?

**Review by KhalidUmuzi — CHANGES_REQUESTED** [review]

File: frontend/src/components/regions/UserActions/index.jsx [hunk 9]
```diff
@@ -6,86 +6,61 @@ import { useParams } from "react-router-dom";
...
-    fetchProjectReviewsPages({
-      dataSequence: [
-        { page: 1, reviewerUser: userId },
-        // { page: 1, recruitUsers: [userId] },
-      ],
-    });
-  }, [fetchProjectReviewsPages, userId]);
-
   useEffect(() => {
     fetchCardCompletions({ page: 1, assigneeUserId: userId });
```
Comment (Ngoakor12) [inline frontend/src/components/regions/UserActions/index.jsx:33]: is this not part of the old activity log aswell?

File: frontend/src/components/regions/UserActions/index.jsx [hunk 10]
```diff
@@ -194,7 +143,7 @@ const mapDispatchToProps = (dispatch) => {
         })
       );
     },
-    
+
     fetchCardCompletions: ({ assigneeUserId, page }) => {
```
Comment (Ngoakor12) [inline frontend/src/components/regions/UserActions/index.jsx:141]: same as this one?

File: frontend/src/components/regions/UserActions/index.jsx [hunk 11]
```diff
@@ -6,86 +6,61 @@ import { useParams } from "react-router-dom";
...
-];
+import { eventTypeColors } from "../../../colors";
+import { addEventColorsToLogEntries, matchEventTypesWithColors } from "./utils";
 
 function UserActionsUnconnected({
   authedUserId,
-  projectReviews,
-  cardSummaries,
-  fetchProjectReviewsPages,
   fetchCardCompletions,
```
Comment (Ngoakor12) [inline frontend/src/components/regions/UserActions/index.jsx:15]: fetchCardCompletions is from the old activity log

**Review by Ngoakor12 — CHANGES_REQUESTED** [review]
Mostly looks good to me...

- Aren't you gonna delete the `DayLog.jsx` since it's part of the old activity log?
- Also... I think the activity log pagination seems to be broken? Or is it it just me? It seems to only fetch the first page to me

File: frontend/src/components/regions/UserActions/index.jsx (hunk 9 above)
Comment (FaithMo) [reply frontend/src/components/regions/UserActions/index.jsx:33]: part of burndown

File: frontend/src/components/regions/UserActions/index.jsx (hunk 11 above)
Comment (FaithMo) [reply frontend/src/components/regions/UserActions/index.jsx:15]: part of burndown

File: frontend/src/components/regions/UserActions/index.jsx (hunk 10 above)
Comment (FaithMo) [reply frontend/src/components/regions/UserActions/index.jsx:141]: part of burndown

File: frontend/src/components/regions/UserActions/Presentation.jsx [hunk 12]
```diff
@@ -3,28 +3,29 @@ import Paper from "@material-ui/core/Paper";
...
   currentUserBurndownStats,
+  activityLogEntries,
+  fetchNextPages,
 }) {
   const classes = useStyles();
   return (
-    <div className={classes.column} onScroll={handleScroll}>
-      <Grid container>
+    <div className={classes.column}>
+      <Grid container onScroll={handleScroll}>
```
Comment (Ngoakor12) [inline frontend/src/components/regions/UserActions/Presentation.jsx]: move the onscroll handler back to where it was so that those who can paginate via scrolling can and those who can't can use the button

**Review by Ngoakor12 — CHANGES_REQUESTED** [review]

**Review by Ngoakor12 — APPROVED** [review]

File: frontend/src/stories/fixtures/activityLogList.json [hunk 13]
```diff
@@ -5,12 +5,20 @@
     "timestamp": "2022-09-02T13:46:16.295020Z",
     "actorUser": 352,
     "effectedUser": 802,
+    "actorUserEmail": "faith.mofokeng@umuzi.org",
+    "effectedUserEmail": "faith.mofokeng@umuzi.org",
     "object1ContentTypeName": "git_real | pull request review",
     "object1Id": 53534,
     "object2ContentTypeName": "git_real | repository",
     "object2Id": 7250,
+    "object_1Summary": {
```
Comment (sheenarbw) [inline frontend/src/stories/fixtures/activityLogList.json]: are you sure it wouldn't be object1Summary?

File: frontend/src/apiAccess/apis.js [hunk 14]
```diff
@@ -405,6 +410,11 @@ async function activityLogEntries({
   if (object1Id) params["object_1_id"] = object1Id;
   if (object2ContentType) params["object_2_content_type"] = object2ContentType;
   if (object2Id) params["object_2_id"] = object2Id;
+  if (timestamp) params["timestamp"] = timestamp;
+  if (actorUserEmail) params["actor_user_email"] = actorUserEmail;
+  if (effectedUserEmail) params["effected_user_email"] = effectedUserEmail;
+  if (object1Summary) params["object_1_summary"] = object1Summary;
+  if (object2Summary) params["object_2_summary"] = object2Summary;
```
Comment (sheenarbw) [inline frontend/src/apiAccess/apis.js]: please double check what filters are implemented on the backend. search for `class ActivityLogEntryViewSet`

File: frontend/src/components/regions/UserActions/ActivityLog.jsx [hunk 15]
```diff
@@ -20,47 +26,41 @@ const useStyles = makeStyles((theme) => ({
...
-              {item.timestamp.substring(0, 10) === timestamp && (
+              {item.timestamp.substring(0, 10) ===
+                timestamp.substring(0, 10) && (
                 <Paper colors={item.eventColor} timestamp={item.timestamp}>
                   <p className={classes.time}>
                     {new Date(item.timestamp).toTimeString().substring(0, 8)}
                   </p>
                   <Typography variant="h6" className={classes.title}>
-                    {item.title}
+                    {item.object_1Summary.title}
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserActions/ActivityLog.jsx]: are you sure about this name?

File: frontend/src/components/regions/UserActions/index.jsx [hunk 16]
```diff
@@ -98,65 +77,35 @@ function UserActionsUnconnected({
...
+    eventTypeColors,
+  });
 
-  let actionLogByDate = {};
-  actionLog.forEach((o) => {
-    const date = o.dateStr;
-    if (orderedDates.indexOf(date) === -1) orderedDates.push(date);
-    actionLogByDate[date] = actionLogByDate[date] || [];
-    actionLogByDate[date].push(o);
+  activityLogEntries = addEventColorsToLogEntries({
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserActions/index.jsx:88]: can't you make one function call instead of two?

`addEventColorsToLogEntries` can call `matchEventTypesWithColors` if it needs to.

File: frontend/src/components/regions/UserActions/index.jsx [hunk 17]
```diff
@@ -170,7 +119,10 @@ const mapStateToProps = (state) => {
       state.FETCH_RECRUIT_PROJECT_REVIEWS_PAGE,
     FETCH_USER_ACTIONS_CARDS_COMPLETED_PAGE:
       state.FETCH_USER_ACTIONS_CARDS_COMPLETED_PAGE,
+    FETCH_ACTIVITY_LOG_ENTRIES: state.FETCH_ACTIVITY_LOG_ENTRIES,
     userBurndownStats: state.apiEntities.burndownSnapshots || {},
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserActions/index.jsx]: I think thios line was meant to be cleaned up in a different pr. It should be 

```
    userBurndownStats: state.apiEntities.burndownSnapshots,
```

Can you please fix it?

**Review by sheenarbw — CHANGES_REQUESTED** [review]
This PR actually covers a few different issues. But you've done a pretty good job. There are a few problems though.

Also, please include a screenshot if you can.

File: frontend/src/stories/fixtures/activityLogList.json (hunk 13 above)
Comment (FaithMo) [reply frontend/src/stories/fixtures/activityLogList.json]: the API returns `object_1Summary`

File: frontend/src/components/regions/UserActions/index.jsx (hunk 17 above)
Comment (FaithMo) [reply frontend/src/components/regions/UserActions/index.jsx]: can i fix this in a different PR? its breaking the burndown chart when I change it

File: frontend/src/apiAccess/apis.js [hunk 18]
```diff
@@ -383,12 +383,17 @@ async function activityLogDayCountsPage({
...
+  actorUserEmail,
+  effectedUserEmail,
   object1ContentType,
   object1Id,
   object2ContentType,
   object2Id,
+  object1Summary,
+  object2Summary,
   page,
 }) {
```
Comment (Ngoakor12) [inline frontend/src/apiAccess/apis.js]: Everything seems to be working as before... just have one question, are we no longer filtering by the new added fields? if not should they not be removed since they're not being used inside that function?

**Review by Ngoakor12 — CHANGES_REQUESTED** [review]

**Review by Ngoakor12 — APPROVED** [review]

File: frontend/src/stories/fixtures/activityLogList.json (hunk 13 above)
Comment (KhalidUmuzi) [reply frontend/src/stories/fixtures/activityLogList.json]: Hmm that is weird. It does do that but why?

**Review by KhalidUmuzi — APPROVED** [review]

File: frontend/src/stories/fixtures/activityLogList.json [hunk 19]
```diff
@@ -5,12 +5,20 @@
...
     "object1Id": 53534,
     "object2ContentTypeName": "git_real | repository",
     "object2Id": 7250,
+    "object1Summary": {
+      "topicProgress": 16,
+      "card": 76,
+      "title": "Understanding the review column",
+      "flavourNames": []
+    },
+    "object_2Summary": null,
```
Comment (Ngoakor12) [inline frontend/src/stories/fixtures/activityLogList.json]: you may not be using `object_2Summary` property but should it not be like `object1Summary`? there are a few instances of `object_2Summary` remaining, for the sake of keeping the data consistent, can you make sure they match

**Review by Ngoakor12 — CHANGES_REQUESTED** [review]

**Review by Ngoakor12 — APPROVED** [review]

File: frontend/src/components/regions/UserActions/index.jsx (hunk 17 above)
Comment (KhalidUmuzi) [reply frontend/src/components/regions/UserActions/index.jsx]: @FaithMo This is a part of the render tidy but has not been prioritized as a task yet. However in the interest of Scout rules, I think you can go ahead and fix this render tidy. Should be a quick one and I think with @Ngoakor12 helping you out, you can hit it out of the park. This is a small exception since This one seems like a really small one to fix.

If you feel it is larger than I'm seeing at a glance then by all means, do it on another branch. If not, its a quick win.

**Review by KhalidUmuzi — CHANGES_REQUESTED** [review]

**Review by Ngoakor12 — APPROVED** [review]

**Review by KhalidUmuzi — APPROVED** [review]

File: frontend/src/apiAccess/helpers.js [hunk 20]
```diff
@@ -8,9 +8,14 @@ export function getNextPageNumberFromUrl({ url }) {
 }
 
 const _toCamel = (s) => {
-  return s.replace(/([-_][a-z])/gi, ($1) => {
-    return $1.toUpperCase().replace("-", "").replace("_", "");
-  });
+  let arr = s.split("");
```
Comment (sheenarbw) [inline frontend/src/apiAccess/helpers.js:11]: This kind of change would have been more easily done by following TDD. Can you please write tests for it?  Also, TDD is a really really good tool to use when bug fixing.

Step 1: write a test that fails because of the bug (red)
Step 2: fix the bug (green)
Step 3: refactor

File: frontend/src/components/regions/UserActions/utils.js [hunk 21]
```diff
@@ -1,3 +1,37 @@
+export function addEventColorsToLogEntries({
+  eventTypes, 
+  eventTypeColors,
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserActions/utils.js]: Can't you just import eventTypeColors? Why is it an argument?

File: frontend/src/components/regions/UserActions/utils.js [hunk 22]
```diff
@@ -1,3 +1,37 @@
...
+  activityLogEntries,
+}) {
+
+  const eventTypesWithEventNames = [];
+  eventTypes = Object.keys(eventTypes).map((key) => eventTypes[key]);
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserActions/utils.js]: Why not: `eventTypesArray = Object.values(eventTypes)`

File: frontend/src/components/regions/UserActions/utils.js [hunk 23]
```diff
@@ -1,3 +1,37 @@
...
+
+  eventTypes.forEach((eventType) => {
+    eventTypesWithEventNames.push({
+      id: eventType.id,
+      eventName: eventType.name,
+      eventColor: eventTypeColors[eventType.name],
+    });
+  });
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserActions/utils.js]: Actually, there are a few things that can be done in a nicer way. How about:
```
eventTypesWithEventNames = Object.values(eventTypes).map(eventType => {
  return {
      id: eventType.id,
      eventName: eventType.name,
      eventColor: eventTypeColors[eventType.name],
}})
```

File: frontend/src/components/regions/UserActions/utils.js [hunk 24]
```diff
@@ -1,3 +1,37 @@
...
+    const data = eventTypesWithEventNames.find(
+      (elem) => elem.id === item.eventType
+    );
+    return {
+      ...item,
+      eventName: data ? data.eventName : "",
+      eventColor: data ? data.eventColor : "",
+    };
+  });
+  return newData;
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserActions/utils.js:28]: all the other functions in this file have tests. Should this have tests too?

File: frontend/src/components/regions/UserActions/index.jsx [hunk 25]
```diff
@@ -210,6 +155,20 @@ const mapDispatchToProps = (dispatch) => {
...
       );
     },
+    fetchActivityLogEntries: ({ actorUser, page }) => {
+      dispatch(
+        apiReduxApps.FETCH_ACTIVITY_LOG_ENTRIES.operations.start({
+          data: { actorUser, page },
+        })
+      );
+    },
+    fetchEventTypes: ({ page }) => {
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserActions/index.jsx]: If this function is only ever called with page =1, why not:

```
fetchEventTypes: ()=>{

dispatch(
        apiReduxApps.FETCH_EVENT_TYPES.operations.start({
          data: { page:1 },
        })
}

...
```

**Review by sheenarbw — CHANGES_REQUESTED** [review]
Getting there

**Review by Ngoakor12 — APPROVED** [review]

**Review by KhalidUmuzi — APPROVED** [review]

**Review by sheenarbw — APPROVED** [review]
